rfcs/text/0008_pulse.md

```diff
@@ -0,0 +1,243 @@
+
+This endpoint will read the `instructions-*` indices, filtering `updated-at` by the `since` query parameter (if provided) and it will return the results, grouping them by channels.
+
+Additionally, we can consider accepting an additional query parameter to retrieve only specific channels. For use cases like distributed components (endpoint, apm, beats, ...) polling instructions themselves.
+
+#### Data model
+
+The storage of each of the documents, will be based on monthly-rolling indices split by channels. This means we'll have indices like `pulse-raw-{CHANNEL_NAME}-YYYY.MM` and `pulse-instructions-{CHANNEL_NAME}-YYYY.MM` (final names TBD).
+
+The first group will be used to index all the incoming documents from the telemetry. While the second one will contain the instructions to be sent to the deployments.
+
+The mapping for those indices will be **`strict`** to avoid anyone storing unwanted/not-allowed info. The indexer defined in [the _Inject telemetry_ endpoint](#inject-telemetry) will need to handle accordingly the errors derived from the strict mapping.  
+We'll set up a process to add new mappings and their descriptions before every new release.
+
+#### Access control
+
+- The access to _raw_ data indices will be very limited. Only granted to those in need of troubleshooting the service and maintaining mappings (this is the Pulse/Telemetry team at the moment).
+- Special views (as in aggregations/visualisations/snapshots of the data stored in special indices via separated indexers/aggregators/ES transform or via _BigQuery_ or similar) will be defined for different roles in the company to help them to take informed decisions based on the data.  
+This way we'll be able to control "who can see what" on a very granual basis. It will also provide us with more flexibility to change to structure of the _raw_ if needed.
+
+### 2. Local Pulse Service
+
+This refers to the plugin running in Kibana in each of our customers' deployments. It will be a core plugin in NP, available for all others plugins to get the existing channels to send pieces of data and subscribe to instructions.
+
+The channel handlers are only defined inside the pulse context and are used to normalise the data for each channel before sending it to the remote service. The CODEOWNERS should notify the Pulse team every time there's an intended change in this context.
+
+#### Data storage
+
+For the purpose of transparency, we want the user to be able to retrieve the telemetry we send at any point, so we should store the information we send for each channel in their own local internal indices (similar to a copy of the `pulse-raw-*` and `pulse-instructions-*` indices in our remote service). In the same effort, we could even provide some _dashboards_ in Kibana for specific roles in the cluster to understand more about their deployment.
```

Comment:
For these local internal indices, will they be "dot indices"? Otherwise, we'll want some kind of acknowledgement from an appropriately authorized end-user to begin creating documents in the "non dot indices" because the user might intend for them to be used for something else or already have roles/users created who have access to these indices.

Reply:
Yes! That's what I meant with _internal indices_. I'll change it to make it more clear.

We should also limit its access to an admin role initially, but let the user grant access to other users that may want to read from it.

I wonder if it's valuable to have a config parameter so the administrator can overwrite the index prefix?